app/models/concerns/with_discussion_creation/subscription.rb

```diff
@@ -8,7 +8,7 @@ module WithDiscussionCreation::Subscription
   end
 
   def subscriptions_in_organization
-    subscriptions.joins(:discussion).where(discussion: discussions_in_organization)
+    subscriptions.where(discussion: Organization.current.discussions_in_organization)
```

Comment:
The `joins` is not needed here.

The original `discussions_in_organization` was evaluating to line 8 here, I believe:

https://github.com/mumuki/mumuki-domain/blob/97a0b79d492d873e296b6f34e5e90407f1da3818/app/models/concerns/with_discussion_creation.rb#L1-L10

However, this would only get discussions where the user was the initiator - I'm not entirely sure if this is an automatic thing because of the relation between User and Discussion and the corresponding FK or what. I'm afraid I don't have this very clear, if I'm honest.

By asking for the organization's `discussions_in_organization` (which delegates that to its book), we do get _all_ the discussions, whether the user was the initiator or not:

https://github.com/mumuki/mumuki-domain/blob/97a0b79d492d873e296b6f34e5e90407f1da3818/app/models/book.rb#L20-L22

We can't just move the method from the book to the organization because it's used in Laboratory's controller to list all discussions.

https://github.com/mumuki/mumuki-laboratory/blob/431c50184478841811925075de969cb59b359a8d/app/controllers/discussions_controller.rb#L65-L67

This way, subscriptions do consider all discussions pertaining to the organization.

Reply:
Also: the way the whole subscription/notification thing works will have to be refactored, as we've talked about before, but I'm trying to fix this bug as cleanly as possible without breaking anything else.

Reply:
Nice catch. It was completely unnecessary. 

> However, this would only get discussions where the user was the initiator - I'm not entirely sure if this is an automatic thing because of the relation between User and Discussion and the corresponding FK 

Yes, it is because of the FK in `discussions`. The `organic_on` simply creates a methods that introduces the organization as a filter.

Reply:
However, this is still obscure to me, mostly because of names. Why does a subscription have discussions? Isn't subscription a marker of interest over _one_ discussion? I don't get it's semantics. Perhaps as you said, we just have to go on with this and fix the issue, but please add a TODO here since this is something that has no sense for me at all. 

What is worst, we are changing implementation of `subscriptions_in_organization` regarding the discussions involved - we are moving to _initiated_ discussions to _participating_ discussions -, but nothing about this change is being reflected in the signature.

Reply:
A subscription does have one discussion only. That `where` is just making a query with a `IN` operator.

```ruby
Subscription.where(discussion_id: [1, 2, 3])
  Subscription Load (1.3ms)  SELECT "subscriptions".*
                             FROM "subscriptions"
                             WHERE "subscriptions"."discussion_id" IN (1, 2, 3)
  ...
```

Just that instead of IDs we're dealing with the discussions themselves, so it's `.where(discussion: ...)`

Regarding the implementation in `subscriptions_in_organization`, I don't really think the signature should change at all. Subscriptions should have always worked with both initiated and participating discussions, in fact, I'd assume they probably did at first and then stopped working without us noticing. I'm positive I'm just fixing a bug, not extending or changing the feature.